pkg/parser/parser.go

```diff
@@ -21,13 +21,40 @@ type packagesCache map[string]*build.Package
 func NewParser(
 	coverProfileFiles []string,
 	logger logrus.FieldLogger,
-) *Parser {
+) (*Parser, error) {
+
+	packagesCache := make(packagesCache)
+	packages := make(map[string]*Package)
+
+	for _, coverFile := range coverProfileFiles {
```

Comment:
not a good idea to put the parsing into the "constructor". 
please separate it into another function.